test/test_foreach.py

```diff
@@ -1114,6 +1115,16 @@ def test_0dim_tensor_overload_exception(self):
         with self.assertRaisesRegex(RuntimeError, "scalar tensor expected to be 0 dim but"):
             torch._foreach_mul(tensors, torch.tensor([1.0, 1.0], device="cuda"))
 
+    @onlyCUDA
+    def test_ignore_stride_mismatch_when_dim_is_0(self):
+        w = torch.nn.Parameter(torch.zeros((2, 1, 100), device='cuda'))
+        x = torch.rand((10, 2, 100), device='cuda')
+        optimizer = optim.AdamW([w], lr=0.0001, fused=True)
+        y = torch.bmm(x.transpose(0, 1), w.transpose(1, 2))
+        y.sum().backward()
+        self.assertNotEqual(w.stride(), w.grad.stride())
+        optimizer.step()
+
```

Comment:
Would be good to also add a test where we expect the error if the stride is mismatched and the size is not 1 to make sure we did not relax the check too much